### PR TITLE
[ID-469]Update the client api to accept authtoken

### PR DESCRIPTION
**Pull request by carmen0208**
# Summary
Update the client API to accept auth token

# Why the changes

# Things worth calling out
* pulled openapi.json from prod, but only added the change needed for PassportImxProvider.
* All the typescript codes are generated.

# Before merging
- [ ] ***For Immutable developers:*** Do the changes in this PR require documentation updates? Please refer to this [SDK documentation guide](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1916994017/SDK+documentation+guide) and list links to documentation update PRs (they don't have to be merged) below (or write `N/A`):
    - [ ] *Add documentation update 1*
    - [ ] *Add documentation update 2*

### CODE REVIEW

**Review by haydenfowler — APPROVED** [review]

File: src/api/domain/orders-api.ts [hunk 1]
```diff
@@ -51,18 +51,15 @@ export const OrdersApiAxiosParamCreator = function (configuration?: Configuratio
...
+         * @param {string} [authorization] Authorization header
          * @param {*} [options] Override http request option.
          * @throws {RequiredError}
          */
-        cancelOrder: async (xImxEthAddress: string, xImxEthSignature: string, id: string, cancelOrderRequest: CancelOrderRequest, options: AxiosRequestConfig = {}): Promise<RequestArgs> => {
-            // verify required parameter 'xImxEthAddress' is not null or undefined
-            assertParamExists('cancelOrder', 'xImxEthAddress', xImxEthAddress)
-            // verify required parameter 'xImxEthSignature' is not null or undefined
-            assertParamExists('cancelOrder', 'xImxEthSignature', xImxEthSignature)
+        cancelOrder: async (id: string, cancelOrderRequest: CancelOrderRequest, xImxEthAddress?: string, xImxEthSignature?: string, authorization?: string, options: AxiosRequestConfig = {}): Promise<RequestArgs> => {
```
Comment (cajames) [inline src/api/domain/orders-api.ts:62]: @carmen0208 introducing the `authorization` param here is a breaking change. We'd need to handle this at the parent workflow.

Comment (carmen0208) [reply src/api/domain/orders-api.ts:62]: How is the breaking change? it is optional?

File: src/api/domain/orders-api.ts [hunk 2]
```diff
@@ -1021,7 +1041,7 @@ export class OrdersApi extends BaseAPI {
      * @memberof OrdersApi
      */
     public cancelOrder(requestParameters: OrdersApiCancelOrderRequest, options?: AxiosRequestConfig) {
-        return OrdersApiFp(this.configuration).cancelOrder(requestParameters.xImxEthAddress, requestParameters.xImxEthSignature, requestParameters.id, requestParameters.cancelOrderRequest, options).then((request) => request(this.axios, this.basePath));
+        return OrdersApiFp(this.configuration).cancelOrder(requestParameters.id, requestParameters.cancelOrderRequest, requestParameters.xImxEthAddress, requestParameters.xImxEthSignature, requestParameters.authorization, options).then((request) => request(this.axios, this.basePath));
```
Comment (carmen0208) [inline src/api/domain/orders-api.ts:1044]: @cajames did you mean to handle it here? it is already generated

File: src/api/domain/orders-api.ts (hunk 1 above)
Comment (carmen0208) [reply src/api/domain/orders-api.ts:62]: @cajames I got what you mean, but the **entrypoint** of the workflow should be in this line: https://github.com/immutable/imx-core-sdk/pull/289/files#diff-2a850f6705a84b936769ce7b1231d25e04b6169b5005a64b31caaa02e16ad6d6R1044

File: src/api/domain/orders-api.ts (hunk 2 above)
Comment (cajames) [reply src/api/domain/orders-api.ts:1044]: Ah got it, yep that makes sense.

**Review by cajames — APPROVED** [review]

**Pull request merged**